pkg/controllers/provisioning/provisioner.go

```diff
@@ -95,20 +94,20 @@ func (p *Provisioner) provision(ctx context.Context) error {
 			pods = append(pods, item.(*v1.Pod))
 		}
 	}
+	logging.FromContext(ctx).Infof("Batched %d provisionable pod(s) of %d pod(s) in %s", len(pods), len(items), window)
```

Comment:
This log line begs the question, "what is this out of". The answer is not satisfying to users, that our cache was stale in one are of the code and not stale in another. 

Here's an option:
```suggestion
        if len(pods) == 0 {
          return nil
        }
	logging.FromContext(ctx).Infof("Batched %d pod(s) in %s", len(pods), window)
```